docs/source/architecture/data_services_architecture.rst

```diff
@@ -1,4 +1,109 @@
-Data Services Architecture
-==========================
+Motech Data Services Architecture
+=================================
 
-There will be text here.
+MOTECH Data Services (MDS) is responsible for creation of Data Models for MOTECH Platform either using Developer Defined Entities (DDE) or End User Defined Entities (EUDE). DDEs use custom annotations of MDS. The EUDEs can be created as well as configured from UI along with their Data Access Layer. Centre of Data Access Layer is Generic Data Access Pattern Class which defines generic persistence related methods.
+
+         .. image:: mds_architecture.png
+            :scale: 100 %
+	  		:alt: Motech Data Services Architecture Diagram
+            :align: center
+
+
+        Section following describes motech data services architecture in figure.
+
+MDS UI Layer
+============
+
+#.Schema Editor: It is responsible for browsing through entities.
+#.Data Browser: It is responsible for browsing through entities.
```

Comment:
From these descriptions, one would think that the Schema Editor and the Data Browser serve the same purpose. Can you elaborate on what they are for?